### ISSUE TITLE
Window Splat for Multiple Monitors

### ISSUE THREAD

**Issue by mogenson** [timbertson/slinger#6]
The window splat feature does not appear to support multiple monitors.

If I have two windows on two monitors like so:
```
+---------------+ +---------------+
|               | |   +-------+   |
|   +-------+   | |   |   B   |   |
|   |   A   |   | |   |       |   |
|   |       |   | |   +-------+   |
|   +-------+   | |               |
+---------------+ +---------------+
```
And press `Super+*` to splat, I assume I would get something like:
```
+---------------+ +---------------+
|---------------| |---------------|
||             || ||             ||
||      A      || ||      B      ||
||             || ||             ||
|---------------| |---------------|
+---------------+ +---------------+
```
 But, instead both windows try to half-tile and end up as:
```
+---------------+ +---------------+
|-------+       | |       +-------|
||      |       | |       |      ||
||  A   |       | |       |  B   ||
||      |       | |       |      ||
|-------+       | |       +-------|
+---------------+ +---------------+
```
This behavior extends to multiple windows as well. Perhaps the best way to support multiple monitors is to treat each monitor as an independent display. Performing an `Super+*` splat should only arrange the windows on the active display, using the extents of the display, without touching any other display.

I'm on X11, Gnome 3.26.

**Comment by jvlatzko** [comment]
This is in fact super-annoying. I use the feature (okay, rather, I'd like to use this feature) extensively due to constantly changing from/to external screens with my laptop, so to me this would be precious ;-) 
@timbertson anything we can do to help out on this?

**Comment by timbertson** [comment]
So talking out loud, the code that implements this is here: https://github.com/timbertson/slinger/blob/bc746323678029ee6a7dfb89420bb486b75f3810/src/actions.ts#L313

Currently it gets the first window's workspace area, and uses that. To support multiple monitors, you'd want to make this into a distributeWindows(area, windows) function Then the main `partition` function would group windows by their workspace / screen (I always get confused at which is which), and call `distributeWindows` for each group. Or, actually, as the issue suggests you could just act on the current screen, which simply means ignoring all windows with a different screen/workspace.

Mainly I'm unfamiliar with exactly what APIs define screen bounds, and it's hard without having multiple monitors myself to figure it out. So I can help with the code logic, but I'd need some help with the screen APIs.

**Comment by timbertson** [comment]
Actually, on second look, the code is already supposed to filter windows for just the current screen. `distribute` (splat) works on `visibleWindows()`, which calls `listWindows(currentWindow)`:

https://github.com/timbertson/slinger/blob/bc746323678029ee6a7dfb89420bb486b75f3810/src/gnome_shell.ts#L185-L195

As you can see from the comment, I wasn't sure whether this would filter out windows on other screens. It's definitely _trying to_, so if you can help figure out the bug then splat should work fine for multiple screens.

**Comment by juarezr** [comment]
Fixed partially with #16.
It works only in the monitor of the focused window. Previously it mixed monitors from current worspace.
For distributing in both monitors one should develop a new shortcut for interating in the monitor list.

**Comment by timbertson** [comment]
Thanks - I'm pretty happy with "just tile the current window". I doubt a shortcut for splatting both monitors is worthwhile.
